sycl/include/CL/sycl/detail/buffer_impl.hpp

```diff
@@ -47,41 +48,47 @@ using cl::sycl::detail::MemoryManager;
 
 template <typename AllocatorT> class buffer_impl : public SYCLMemObjT {
 public:
-  buffer_impl(size_t SizeInBytes, const property_list &PropList,
+  buffer_impl(size_t SizeInBytes, size_t SizeOfType,
+              const property_list &PropList,
               AllocatorT Allocator = AllocatorT())
-      : buffer_impl((void *)nullptr, SizeInBytes, PropList, Allocator) {}
+      : buffer_impl((void *)nullptr, SizeInBytes, SizeOfType, PropList,
+                    Allocator) {}
 
-  buffer_impl(void *HostData, size_t SizeInBytes, const property_list &Props,
-              AllocatorT Allocator = AllocatorT())
+  buffer_impl(void *HostData, size_t SizeInBytes, size_t SizeOfType,
+              const property_list &Props, AllocatorT Allocator = AllocatorT())
       : MSizeInBytes(SizeInBytes), MProps(Props), MAllocator(Allocator) {
 
     if (!HostData)
       return;
 
     set_final_data(reinterpret_cast<char *>(HostData));
-    if (MProps.has_property<property::buffer::use_host_ptr>()) {
+    const size_t RequiredAlignment = getNextPowerOfTwo(SizeOfType);
```

Comment:
This is supposed to be compile time known value.
Can we make use of C++ `constexpr` to avoid evaluation of `getNextPowerOfTwo` at runtime?

Reply:
Done